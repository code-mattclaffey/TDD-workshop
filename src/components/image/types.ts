import { GenericComponent } from '../../module';

export interface ImageInterface extends GenericComponent {
    src: string;
    alt: string;
}
