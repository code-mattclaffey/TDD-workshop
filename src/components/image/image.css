/* ------------------------------------
    #IMAGES
------------------------------------*/

img,
.e-img {
    height: auto;
    max-width: 100%;
}
