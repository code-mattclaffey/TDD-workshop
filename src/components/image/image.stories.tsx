import React from 'react';
import { Image } from '.';

export default {
    title: 'Image',
    componentSubtitle: '',
};

export const image = () => (
    <Image
        src="https://via.placeholder.com/400x300"
        alt="A placeholder image"
    />
);
