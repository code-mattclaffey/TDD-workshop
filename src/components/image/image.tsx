import React from 'react';
import classNames from 'classnames';
import { ImageInterface } from './types';

export const Image = ({
    additionalClassNames = '',
    src,
    alt,
    children,
    ...rest
}: ImageInterface) => {
    const classes = classNames(additionalClassNames);

    return <img className={classes} {...rest} src={src} alt={alt} />;
};
