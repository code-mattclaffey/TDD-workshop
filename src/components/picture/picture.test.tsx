import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Picture, Source } from '.';
import { Image } from '../image';

afterEach(() => {
    cleanup();
});

describe('<Picture />', () => {
    it('should match snapshot', () => {
        const { container } = render(
            <Picture width={300} height={400}>
                <Source
                    srcSet="https://via.placeholder.com/400x300"
                    media="(max-width: 800px)"
                />
                <Image
                    src="https://via.placeholder.com/400x300"
                    alt="A placeholder image"
                />
            </Picture>,
        );

        expect(container).toMatchSnapshot();
    });
});
