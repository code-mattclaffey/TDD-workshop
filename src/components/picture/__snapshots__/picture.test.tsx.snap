// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Picture /> should match snapshot 1`] = `
<div>
  <picture
    class="e-picture"
    style="padding-top: 133.33333333333331%;"
  >
    <source
      media="(max-width: 800px)"
      srcset="https://via.placeholder.com/400x300"
    />
    <img
      alt="A placeholder image"
      class="e-img"
      src="https://via.placeholder.com/400x300"
    />
  </picture>
</div>
`;
